src/AggregateFunctions/AggregateFunctionSparkbar.h

```diff
@@ -100,8 +89,9 @@ class AggregateFunctionSparkbar final
 
 private:
     size_t width;
-    X min_x;
-    X max_x;
+    mutable X min_x;
+    mutable X max_x;
+    bool specified_min_max_x;
```

Comment:
Looks like it's better to store it in aggregate function data, but not in function itself. Is there any reason to put it here?

Reply:
If ```min_x```, ```max_x``` are not specified, we need to generate a global ```min_x```, ```max_x``` based on the data, and the later ```render```  method will rely on them to do sparkbar alignment.

Reply:
Why we can't do it with `AggregateFunctionSparkbarData`? Storing data in function itself ma cause issues if query would be distributed.

Reply:
You're right.